sparse.h

```diff
@@ -0,0 +1,48 @@
+#ifndef __SPARSE_H__
+#define __SPARSE_H__
+
+#include <stdint.h>
+
+typedef struct __attribute__((__packed__)) sparse_header {
+	uint32_t magic; /* 0xed26ff3a */
+	uint16_t major_version; /* (0x1) - reject images with higher major versions */
+	uint16_t minor_version; /* (0x0) - allow images with higer minor versions */
+	uint16_t file_hdr_sz; /* 28 bytes for first revision of the file format */
+	uint16_t chunk_hdr_sz; /* 12 bytes for first revision of the file format */
+	uint32_t blk_sz; /* block size in bytes, must be a multiple of 4 (4096) */
+	uint32_t total_blks; /* total blocks in the non-sparse output image */
+	uint32_t total_chunks; /* total chunks in the sparse input image */
+	uint32_t image_checksum; /* CRC32 checksum of the original data, counting "don't care" */
+	                         /* as 0. Standard 802.3 polynomial, use a Public Domain */
+	                         /* table implementation */
+} sparse_header_t;
+
+#define SPARSE_HEADER_MAGIC     0xed26ff3a
+#define SPARSE_HEADER_MAJOR_VER 0x0001
+#define SPARSE_HEADER_MINOR_VER 0x0000
+
+typedef struct __attribute__((__packed__)) chunk_header {
+	uint16_t chunk_type; /* 0xCAC1 -> raw; 0xCAC2 -> fill; 0xCAC3 -> don't care */
```

Comment:
thinking out loud; would an enum be better ?

Reply:
this is not a good idea because the compiler does not guarantee the size of enum, and using different optimization techniques enum can be 4 bytes or 1 byte, but never 2 bytes.

Reply:
No problem!